### ISSUE TITLE
arrow-body-style: ["error", "as-needed"] not respecting other rules

### ISSUE THREAD

**Issue by onemen** [eslint/eslint#5775]
**What version are you using?**
2.6.0

**The actual source code that is causing the issue**
```js
var x1 = a => {
   return 1 ? 2 : 3;
};

var x2 = a => 1 ? 2 : 3;

var x3 = a => (1 ? 2 : 3);

var x4 = a => {
  return a.prop == 'check_if_a.prop_is_very_long_string' &&
    a.value == 'check_if_a.value_is_very_long_string';
};

var x5 = a => a.prop == 'check_if_a.prop_is_very_long_string' && a.value == 'check_if_a.value_is_very_long_string';
```

**My configuration**
```js
{
  "root": true,
  "env": {
    "browser": true,
    "es6": true
  },

  "parserOptions": {
    "ecmaVersion": 6
  },

  "rules": {
    "arrow-body-style": ["error", "as-needed"],
    "no-confusing-arrow": ["error", {"allowParens": true}],
    "no-extra-parens": [2, "all"],
    "max-len": ["error", {"code": 80, "tabWidth": 2, "ignoreUrls": true}]
  }
}
```
**The complete and actual output from ESLint**
```sh
C:\test eslint\test.js: line 1, col 15, Error - Unexpected block statement surrounding arrow body. (arrow-body-style)
C:\test eslint\test.js: line 5, col 10, Error - Arrow function used ambiguously with a conditional expression. (no-confusing-arrow)
C:\test eslint\test.js: line 7, col 15, Error - Gratuitous parentheses around expression. (no-extra-parens)
C:\test eslint\test.js: line 9, col 15, Error - Unexpected block statement surrounding arrow body. (arrow-body-style)
C:\test eslint\test.js: line 14, col 1, Error - Line 14 exceeds the maximum line length of 80. (max-len)

5 problems
```
**At the moment I am forced to use:**
```js
var x6 = function(a) {
  return 1 ? 2 : 3;
};

var x7 = function(a) {
  return a.prop == 'check_if_a.prop_is_very_long_string' &&
    a.value == 'check_if_a.value_is_very_long_string';
};
```
**What did you expect to happen?**
The rule 'arrow-body-style' should not report an error for function x1, when the one line version function x2, is restricted by 'no-confusing-arrow'.
Notice that since 'no-extra-parens' report an error for function x3, it look like that no valid syntax will is satisfying all the rules mention above.

The same applies to function x4 and x5 respectfully. The rule 'arrow-body-style' should not report an error for function x4, when the one line version function x5, is restricted by 'max-len'.

**Comment by alberto** [comment]
For the first case, it's a duplicate of [this issue](https://github.com/eslint/eslint/issues/5332).

For the second one, you should write it as:

```
var x4 = a => a.prop == 'check_if_a.prop_is_very_long_string' &&
    a.value == 'check_if_a.value_is_very_long_string';
```

Otherwise, you should use a [eslint comment](http://eslint.org/docs/user-guide/configuring#configuring-rules) to disable the error for that portion of code.
